modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -653,10 +658,21 @@ struct ReadlineContext {
         if(context) {
 #ifdef USE_READLINE
             if(!hist_file.empty()) {
+                // Allow history file size to be limited. <0 or non-numeric value means infinite size, >=0 means truncate to that many lines
+                epicsInt32 maxLines = -1;
+                if (const char* env = getenv("EPICS_IOCSH_HISTFILESIZE"))
+                    epicsParseInt32(env, &maxLines, 10, NULL);
```

Comment:
Need to handle parse errors.  eg. test with `EPICS_IOCSH_HISTFILESIZE=10o`, which should warn and use the default, not `10`.